src/System.Text.Json/tests/JsonNullTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonNullTests
+    {
+        [Fact]
+        public static void TestToString()
+        {
+            Assert.Equal("null", new JsonNull().ToString());
+        }
+
+        [Fact]
+        public static void TestEquals()
+        {
+            var jsonNull1 = new JsonNull();
+            var jsonNull2 = new JsonNull();
+
+            Assert.True(jsonNull1.Equals(jsonNull2));
+            Assert.True(jsonNull1 == jsonNull2);
+            Assert.False(jsonNull1 != jsonNull2);
```

Comment:
I am not sure what you mean. Should the last line be:
```csharp
Assert.True(jsonString1 == jsonString2);
```

Yes, `new JsonString("1")` is equal to `new JsonString("1")`, because of overloaded equals comparing values. For JsonNumbers I am comparing actual value stored as string, so for example `new JsonNumber("1")` is equal to `new JsonNumber("1")` as well.

Reply:
> Should the last line be

Yep, that was a typo.

The behavior makes sense to me. Thanks for clarifying.